Objects/unicodeobject.c

```diff
@@ -3996,6 +3996,40 @@ PyUnicode_FSDecoder(PyObject* arg, void* addr)
 }
 
 
+int
+PyUnicode_GetUTF8Buffer(PyObject *unicode, const char *errors,
+                        Py_buffer *view)
+{
+    if (!PyUnicode_Check(unicode)) {
+        PyErr_BadArgument();
+        return -1;
+    }
+    if (PyUnicode_READY(unicode) == -1) {
+        return -1;
+    }
+
+    if (PyUnicode_UTF8(unicode) != NULL) {
+        return PyBuffer_FillInfo(view, unicode,
```

Comment:
We should check that `Py_TYPE(unicode)->tp_as_buffer == NULL`. Just for the case if the string subclass implement the buffer protocol.

Reply:
What happens if Py_TYPE(unicode)->tp_as_buffer != NULL? It doesn't seem to be used.

Reply:
`Py_TYPE(unicode)->tp_as_buffer` is called when releasing the buffer.